### ISSUE TITLE
[Flaky Test] TaskFilter.filtered_tasks when filtering by regional office

### ISSUE THREAD

**Issue by ferristseng** [department-of-veterans-affairs/caseflow#13454]
## Background/context/resources

```
TaskFilter.filtered_tasks when filtering by regional office when filter includes Boston returns only tasks where the closest regional office is Boston
     Failure/Error: expect(subject.map(&:id)).to match_array(boston_tasks.map(&:id))

       expected collection contained:  [269, 270, 271]
       actual collection contained:    [266, 267, 268]
       the missing elements were:      [269, 270, 271]
       the extra elements were:        [266, 267, 268]
     # ./spec/models/task_filter_spec.rb:176:in `block (5 levels) in <top (required)>'
```

 - Circle CI Error:  https://app.circleci.com/jobs/github/department-of-veterans-affairs/caseflow/103171
 - Has the test already been skipped in the code?
   - [ ] Skipped
   - [ ] Not Skipped
 - Related Flakes
    + N/A

## Approach

Time box this investigation and fix.

Remember that if a test has been skipped for a decent amount of time, it may no longer map to the exact code.

If you reach the end of your time box and don't feel like the solution is in sight:
  - [ ] document the work you've done, including dead ends and research
  - [ ] skip the test in the code
  - [ ] file a follow on ticket
  - [ ] close this issue

**Comment by pkarman** [comment]
with `ENSURE_STABLE=100` I can make this fail locally about 5% of the time.

The bug seems to be that the `boston_tasks` are not always reliably set to `Boston` in the `cached_appeal_attributes` table.

**Comment by pkarman** [comment]
fix in #13896 where I ran `ENSURE_STABLE=1000` with zero errors.
